src/Synchronizer/VariantConfigurationSynchronizer.php

```diff
@@ -22,27 +24,45 @@ final class VariantConfigurationSynchronizer implements VariantConfigurationSync
 
     private VariantCollectionInterface $variantCollection;
 
+    private VariantGeneratorRegistryInterface $generatorRegistry;
+
     public function __construct(
         FactoryInterface $variantConfigurationFactory,
         VariantConfigurationRepositoryInterface $variantConfigurationRepository,
         VariantCollectionInterface $variantCollection,
-        ManagerRegistry $managerRegistry
+        ManagerRegistry $managerRegistry,
+        VariantGeneratorRegistryInterface $generatorRegistry
     ) {
         $this->variantConfigurationFactory = $variantConfigurationFactory;
         $this->variantConfigurationRepository = $variantConfigurationRepository;
         $this->variantCollection = $variantCollection;
         $this->managerRegistry = $managerRegistry;
+        $this->generatorRegistry = $generatorRegistry;
     }
 
-    public function synchronize(): void
+    public function synchronize(bool $runSetup): VariantConfigurationSynchronizationResultInterface
     {
+        $syncResult = new VariantConfigurationSynchronizationResult();
+
         $variantConfiguration = $this->variantConfigurationRepository->findNewest();
         if (null !== $variantConfiguration) {
             $variantCollection = $variantConfiguration->getVariantCollection();
             Assert::notNull($variantCollection);
 
             if ($variantCollection->equals($this->variantCollection)) {
-                return;
+                $syncResult->addMessage('VariantCollection', 'Variant configuration has not changed. No synchronization performed');
+                // TODO: Should runSetup be possible even if the variant collections are the same?
```

Comment:
Yes. We don't know what people want to do in the setup, so when you explicitly say you want to run setup, I think we should

Reply:
Will make that change.

On a related note - should we make it possible to force a new variantCollection to be saved (even if it equals)? It could be a poor solution to some of the problems we foresee with #20 but a solution nonetheless.

Reply:
Maybe, but that would be another issue/PR